plugins/modules/ndo_synce_interface_policy.py

```diff
@@ -0,0 +1,277 @@
+    argument_spec.update(
+        dict(
+            template=dict(type="str", required=True),
+            interface_policy=dict(type="str", aliases=["name"]),
+            interface_policy_uuid=dict(type="str", aliases=["uuid"]),
+            description=dict(type="str"),
+            admin_state=dict(type="str", choices=["enabled", "disabled"], default="disabled"),
+            sync_state_msg=dict(type="str", choices=["enabled", "disabled"], default="disabled"),
+            selection_input=dict(type="str", choices=["enabled", "disabled"], default="disabled"),
+            src_priority=dict(type="int", default=100),
+            wait_to_restore=dict(type="int", default=5),
+            state=dict(type="str", choices=["absent", "query", "present"], default="query"),
+        )
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "present", ["interface_policy"]],
+            ["state", "absent", ["interface_policy"]],
+        ]
+    )
+
+    mso = MSOModule(module)
+
+    template = module.params.get("template")
+    interface_policy = module.params.get("interface_policy")
+    interface_policy_uuid = module.params.get("interface_policy_uuid")
+    description = module.params.get("description")
+    admin_state = module.params.get("admin_state")
+    sync_state_msg = module.params.get("sync_state_msg")
+    selection_input = module.params.get("selection_input")
+    src_priority = module.params.get("src_priority")
+    wait_to_restore = module.params.get("wait_to_restore")
+    state = module.params.get("state")
+
+    ops = []
+    match = None
+
+    mso_template = MSOTemplate(mso, "fabric_policy", template)
+    mso_template.validate_template("fabricPolicy")
+
+    path = "/fabricPolicyTemplate/template/syncEthIntfPolicies"
+
+    existing_interface_policies = mso_template.template.get("fabricPolicyTemplate", {}).get("template", {}).get("syncEthIntfPolicies", [])
+
+    if interface_policy:
+        object_description = "SyncE Interface Policy"
+        if interface_policy_uuid:
+            match = mso_template.get_object_by_uuid(object_description, existing_interface_policies, interface_policy_uuid)
+        else:
+            kv_list = [KVPair("name", interface_policy)]
+            match = mso_template.get_object_by_key_value_pairs(object_description, existing_interface_policies, kv_list)
+        if match:
+            mso.existing = mso.previous = copy.deepcopy(match.details)
+    else:
+        mso.existing = mso.previous = existing_interface_policies
+
+    if state == "present":
+
+        mso.existing = {}
+
+        if match:
+            sync_state_msg_value = ENABLED_OR_DISABLED_TO_BOOL_STRING_MAP.get(sync_state_msg)
+            selection_input_value = ENABLED_OR_DISABLED_TO_BOOL_STRING_MAP.get(selection_input)
+
+            if interface_policy and match.details.get("name") != interface_policy:
```

Comment:
The system will not allow you to change the policy name. Please check the condition once again.

Reply:
Yes, you are correct! I missed to check that part.